File: src/typescript/src/pipeline_services/examples/example03_runtime_pipeline.ts
import { RuntimePipeline } from "../../index.js";
import { normalize_whitespace, strip } from "./text_steps.js";

async function main(): Promise<void> {
  const runtimePipeline = new RuntimePipeline<string>(
    "example03_runtime_pipeline",
    false,
    "  Hello   Runtime  ",
  );
  await runtimePipeline.addAction(strip);
  await runtimePipeline.addAction(normalize_whitespace);
  // eslint-disable-next-line no-console
  console.log("runtimeValue=", runtimePipeline.value());

  const frozenPipeline = runtimePipeline.freeze();
  const output = await frozenPipeline.run("  Hello   Frozen  ");
  // eslint-disable-next-line no-console
  console.log("frozenValue=", output);
}

void main();

File: src/Python/pipeline_services/examples/example03_runtime_pipeline.py
from __future__ import annotations

from pipeline_services.core.runtime_pipeline import RuntimePipeline
from pipeline_services.examples.text_steps import normalize_whitespace, strip


def main() -> None:
    runtime_pipeline = RuntimePipeline("example03_runtime_pipeline", False, "  Hello   Runtime  ")
    runtime_pipeline.add_action(strip)
    runtime_pipeline.add_action(normalize_whitespace)
    print("runtimeValue=", runtime_pipeline.value())

    frozen_pipeline = runtime_pipeline.freeze()
    result = frozen_pipeline.run("  Hello   Frozen  ")
    print("frozenValue=", result.context)


if __name__ == "__main__":
    main()
